adapters/rubicon/rubicon.go

```diff
@@ -964,6 +1030,73 @@ func updateExtWithIabAttribute(target json.RawMessage, data []openrtb2.Data, seg
 	return jsonTarget, nil
 }
 
+func populateFirstPartyDataAttributes(source json.RawMessage, target json.RawMessage) json.RawMessage {
+	targetAsMap := messageToMap(target)
+	sourceAsMap := messageToMap(source)
+
+	for key, val := range sourceAsMap {
+		switch typedValue := val.(type) {
+		case string:
+			targetAsMap[key] = [1]string{typedValue}
+		case float64:
+			if typedValue == float64(int(typedValue)) {
+				targetAsMap[key] = [1]string{strconv.Itoa(int(typedValue))}
+			}
+		case bool:
+			targetAsMap[key] = [1]string{strconv.FormatBool(typedValue)}
+		case []interface{}:
+			if isStringArray(typedValue) {
+				targetAsMap[key] = typedValue
+			}
+			if isBoolArray(typedValue) {
+				targetAsMap[key] = convertToStringArray(typedValue)
+			}
+		}
+	}
+
+	updatedTarget, _ := json.Marshal(targetAsMap)
+	return updatedTarget
+}
+
+func isStringArray(array []interface{}) bool {
+	for _, val := range array {
+		switch val.(type) {
+		case string:
+			continue
+		default:
+			return false
+		}
+	}
+	return true
+}
+
+func isBoolArray(array []interface{}) bool {
+	for _, val := range array {
+		switch val.(type) {
+		case bool:
+			continue
+		default:
+			return false
+		}
+	}
+	return true
+}
+
+func convertToStringArray(arr []interface{}) []string {
+	var stringArray []string
+	for _, val := range arr {
+		stringArray = append(stringArray, strconv.FormatBool(val.(bool)))
+	}
+	return stringArray
+}
+
+func messageToMap(message json.RawMessage) map[string]interface{} {
+	targetAsMap := make(map[string]interface{})
+	_ = json.Unmarshal(message, &targetAsMap)
```

Comment:
This is was done, because `message ` can be nil and this will cause an error, and by ignoring this err I will just return empty map

Reply:
> because message can be nil and this will cause an error

This is exactly why we need to add proper error handling. I understand that you just want to return an empty map when that happens but you can achieve that with proper error handling as well. We require that these errors aren't ignored because otherwise it becomes really difficult to debug (for the Prebid Server host and for the publisher) if an issue arises. Clear error messages on the other hand help properly communicate what exactly went wrong and eliminates guess work. Plus, it also saves a lot of time while debugging.